File: WebSocketChat.Frontend/src/app/shared/models/chat/messages/room-join-request.ts
import { MessageType } from 'app/shared/models/chat/message-type';
import { MessageIdentifier, Message } from './../message';

export class RoomJoinRequest extends MessageIdentifier implements Message  {

    constructor(isSuccessful?:boolean) { super(); }

    messageType : MessageType = MessageType.RoomJoinRequest;

    /**The requested room's identifier */
    roomIdentifier: number;

    /**Supplied password (if it's needed) */
    password: string;

    /**This boolean indicates whether the join-request was successful */
    isSuccessful: boolean;
}

File: WebSocketChat.Server/WebSocketChat.Server/Models/Chat/Messages/RoomJoinRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebSocketChat.Server.Models.Chat.Messages
{
    public class RoomJoinRequest : Message
    {
        public RoomJoinRequest()
        {
            MessageType = MessageType.RoomJoinRequest;
        }

        /// <summary>
        /// The requested room's identifier
        /// </summary>
        public int RoomIdentifier { get; set; }

        /// <summary>
        /// Supplied password (if it's needed)
        /// </summary>
        public string Password { get; set; }

        /// <summary>
        /// This boolean indicates whether the join-request was successful
        /// </summary>
        public bool IsSuccessful { get; set; }
    }
}
